src/libraries/System.Net.Sockets/tests/FunctionalTests/UdpClientTest.cs

```diff
@@ -598,31 +628,71 @@ public async Task SendAsync_ReceiveAsync_Success(bool ipv4)
             using (var sender = new UdpClient(new IPEndPoint(address, 0)))
             {
                 await sender.SendAsync(new byte[1], 1, new IPEndPoint(address, ((IPEndPoint)receiver.Client.LocalEndPoint).Port));
+				await AssertReceiveAsync(receiver);
+				
+				await sender.SendAsync(new ReadOnlyMemory<byte>(new byte[1]), new IPEndPoint(address, ((IPEndPoint)receiver.Client.LocalEndPoint).Port));
+				await AssertReceiveAsync(receiver);
+            }
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        [PlatformSpecific(TestPlatforms.Windows)] // "localhost" resolves to IPv4 & IPV6 on Windows, but may resolve to only one of those on Unix
+        public async Task SendAsync_ReceiveAsync_With_HostName_Success(bool ipv4)
+        {
+            IPAddress address = ipv4 ? IPAddress.Loopback : IPAddress.IPv6Loopback;
 
-                UdpReceiveResult result = await receiver.ReceiveAsync();
-                Assert.NotNull(result.RemoteEndPoint);
-                Assert.NotNull(result.Buffer);
-                Assert.InRange(result.Buffer.Length, 1, int.MaxValue);
+            using (var receiver = new UdpClient(new IPEndPoint(address, 0)))
+            using (var sender = new UdpClient(new IPEndPoint(address, 0)))
+            {
+                await sender.SendAsync(new byte[1], "localhost", ((IPEndPoint)receiver.Client.LocalEndPoint).Port);
+                await AssertReceiveAsync(receiver);
+
+                await sender.SendAsync(new ReadOnlyMemory<byte>(new byte[1]), "localhost", ((IPEndPoint)receiver.Client.LocalEndPoint).Port);
+                await AssertReceiveAsync(receiver);
+            }
+        }
```

Comment:
Let's add one more test for `SendAsync`:

Add `SendAsync_PreCanceled_Throws` test case to make sure the cancellation token is propagated to the socket for `SendAsync` overloads.

Reply:
Added.